c++/src/kj/common.h

```diff
@@ -1928,6 +1928,20 @@ class ArrayPtr: public DisallowConstCopyIfNotConst<T> {
     return true;
   }
 
+  //Lexographic difference function can be overriden for the type by defining <=> on your type
+  //Designed infer the size to compare as the minimum i.e. auto truncates the comparison
+  inline int operator<=>(const ArrayPtr& other) const {
```

Comment:
tests please

Reply:
Please let me know if changes to: c++/src/kj/common-test.c++ are sufficient for this now.